electrumutils.py

```diff
@@ -0,0 +1,212 @@
+import logging
+import binascii
+from utils import BITCOIND_CONFIG
+import time
+import sys
+import os
+import tempfile
+import traceback
+import asyncio
+from electrumx.server.controller import Controller
+from electrumx.server.env import Env
+import threading
+from electrum.daemon import Daemon, get_fd_or_server
+from electrum import simple_config, constants, keystore, util
+from electrum.wallet import Wallet
+from electrum.storage import WalletStorage
+from electrum.mnemonic import Mnemonic
+
+bh2u = lambda x: binascii.hexlify(x).decode('ascii')
+
+class ElectrumX:
+    def __init__(self):
+        print("server/__init__")
+        self.loop = asyncio.new_event_loop()
+        self.evt = asyncio.Event(loop=self.loop)
+
+    def kill(self):
+        print("server/kill")
+        async def set_evt():
+            self.evt.set()
+        asyncio.run_coroutine_threadsafe(set_evt(), self.loop)
+
+    def start(self):
+        print("server/start")
+        from os import environ
+        rpc_port = 8000
+        os.system("rm -rf COIN hist meta utxo")
+        environ.update({
+            "COIN": "BitcoinSegwit",
+            "SSL_KEYFILE": "/home/janus/electrumx/key.pem",
+            "SSL_CERTFILE": "/home/janus/electrumx/cert.pem",
+            "TCP_PORT": "51001",
+            "SSL_PORT": "51002",
+            "RPC_PORT": str(8000),
+            "NET": "regtest",
+            "DAEMON_URL": "http://rpcuser:rpcpass@127.0.0.1:" + str(BITCOIND_CONFIG.get('rpcport', 18332)),
```

Comment:
I recently broke this, by making all ports variable in order to enable parallel test execution, sorry about that. Could you use the `ephemeral_port_reserve.reserve` function to self-assign a unique port?